Chap_API_Tools.tex

```diff
@@ -1338,6 +1344,57 @@ \subsection{\code{PMIx_tool_set_server}}
 Designate the specified server to be the tool's \emph{primary} server for all subsequent \ac{API} calls.
 
 
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+\subsection{\code{PMIx_tool_check_connection}}
+\declareapi{PMIx_tool_check_connection}
+
+%%%%
+\summary
+
+Check if a specified process has connected with the requesting tool.
```

Comment:
I'm not sure I follow this description. The API indicates that the calling tool is checking its connection with the specified server.

I assume that they still need to call `PMIx_tool_attach_to_server` to connect with this server. Or `PMIx_tool_set_server` to switch the connection to this server. The former, at least, should not return until there is a successful connection. What is the use case for this new API then?

Reply:
Imagine the case where a tool fork/exec's an intermediate launcher (IL). In this case, the tool needs to know when the IL has connected back to the tool so the tool can set the IL as its server and then issue directives such as calling `PMIx_Spawn`. Ideally, it would be nice to encapsulate that inside the tool's call to `PMIx_Spawn` as the Standard currently reads. However, that is proving difficult without creating another progress thread. This is an alternate solution.

I suppose another option is to move the delay for connection into "set_server". Let me try and see if there is a way to make that work. We still might want this API as a way of checking the state of a connection (since a tool doesn't have the server's upcall module functions to tell it when a server disappears), but we can hold on that for now.

Reply:
Adding the delay to `PMIx_tool_attach_to_server` makes sense to me - to wait for the designated server to come online. Additionally adding it to `PMIx_tool_set_server` to wait for the confirmation from the server seems good too since we might be switching to this server, and we need to make sure it is still alive.

The new API might be useful if you want to do a liveness check on the server, but I would expect it to return a boolean state given its name (`PMIx_tool_check_connection`). If the server disappears we should get an event in the tool (if we registered for it), right? I would expect the Tool-side PMIx library to throw that event.